### PR TITLE
Support ANNS statement execution

### PR DESCRIPTION
**Pull request by longjiquan**
/kind improvement
With this pr, below statement can be executed by the Milvus and Milvus will return the standard sql result to the MySql client.

```sql
select query_number, pk, random, distance
from hello_milvus
where random > 0.5
anns by embeddings -> ([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8], [0.8, 0.7, 0.6, 0.6, 0.4, 0.3, 0.2, 0.1])
params = (metric_type=L2, nprobe=10)
limit 3;
```

### CODE REVIEW

**Comment by longjiquan** [comment]
image

File: internal/mysqld/executor/executor.go [hunk 1]
```diff
@@ -129,6 +140,242 @@ func (e *defaultExecutor) execSelect(ctx context.Context, n *planner.NodeSelectS
...
+
+	if res.GetStatus().GetErrorCode() != commonpb.ErrorCode_Success {
+		return nil, common.NewStatusError(res.GetStatus().GetErrorCode(), res.GetStatus().GetReason())
+	}
+
+	return wrapSearchResult(res, outputsIndex, userOutputs)
+}
+
+func wrapSearchResult(res *milvuspb.SearchResults, outputsIndex map[string]int, userOutputs []string) (*sqltypes.Result, error) {
+	qnIndex := outputsIndex["query_number"]
```
Comment (czs007) [inline internal/mysqld/executor/executor.go:209]: by design , query_number is also optional, just like distance.
This part can be discussed again.

File: internal/mysqld/executor/executor.go [hunk 2]
```diff
@@ -129,6 +140,242 @@ func (e *defaultExecutor) execSelect(ctx context.Context, n *planner.NodeSelectS
...
+	disIndex, ok := outputsIndex["distance"]
+
+	fieldsData := make([]*schemapb.FieldData, len(res.GetResults().GetFieldsData()))
+	copy(fieldsData, res.GetResults().GetFieldsData())
+
+	qnFieldData := generateQueryNumberFieldData(res)
```
Comment (czs007) [inline internal/mysqld/executor/executor.go:215]: by design , query_number is also optional, just like distance.
This part can be discussed again.

File: internal/mysqld/planner/float_vector.go [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
 type NodeFloatVector struct {
 	Array []float32
 }
 
+func (n NodeFloatVector) Serialize() []byte {
+	data := make([]byte, 0, 4*len(n.Array)) // float32 occupies 4 bytes
+	buf := make([]byte, 4)
+	for _, f := range n.Array {
+		// TODO
+		binary.LittleEndian.PutUint32(buf, math.Float32bits(f))
```
Comment (czs007) [inline internal/mysqld/planner/float_vector.go:17]: use common.Endian

File: internal/mysqld/executor/executor.go [hunk 4]
```diff
@@ -129,6 +140,242 @@ func (e *defaultExecutor) execSelect(ctx context.Context, n *planner.NodeSelectS
...
+
+	ph.Type = placeHolderType
+	for _, vector := range vectors {
+		ph.Values = append(ph.Values, vector.Serialize())
+	}
+
+	return ph
+}
+
+func mapKvPairs(kvs map[string]string) []*commonpb.KeyValuePair {
```
Comment (czs007) [inline internal/mysqld/executor/executor.go:368]: same to funcutil.Map2KeyValuePair

File: internal/mysqld/executor/executor.go [hunk 5]
```diff
@@ -129,6 +140,242 @@ func (e *defaultExecutor) execSelect(ctx context.Context, n *planner.NodeSelectS
...
+	users = make([]string, 0, len(outputs))
+
+	fix := func(s string) string {
+		return strings.ToLower(strings.TrimSpace(s))
+	}
+
+	for i, f := range outputs {
+		fixed := fix(f)
+		index[fixed] = i
+		if fixed != "query_number" && fixed != "distance" {
```
Comment (czs007) [inline internal/mysqld/executor/executor.go:322]: better to change these 2 keywords to constant definitions

File: internal/mysqld/executor/executor.go [hunk 6]
```diff
@@ -129,6 +140,242 @@ func (e *defaultExecutor) execSelect(ctx context.Context, n *planner.NodeSelectS
...
+		for k, v := range annsClause.Params.Unwrap().KVs {
+			searchParams[k] = v
+		}
+		params, _ := json.Marshal(annsClause.Params.Unwrap().KVs)
+		searchParams["params"] = string(params)
+	}
+
+	outputsIndex, userOutputs := generateOutputsIndex(outputs)
+
+	_, queryNumberExist := outputsIndex["query_number"]
```
Comment (czs007) [inline internal/mysqld/executor/executor.go:168]: query_number is optional.

File: internal/mysqld/executor/executor.go [hunk 7]
```diff
@@ -129,6 +140,242 @@ func (e *defaultExecutor) execSelect(ctx context.Context, n *planner.NodeSelectS
...
+	if !q.Limit.IsSome() {
+		return nil, fmt.Errorf("limit not specified in the ANNS statement")
+	}
+	limit := q.Limit.Unwrap().Limit
+	offset := q.Limit.Unwrap().Offset
+
+	annsClause := q.Anns.Unwrap()
+
+	searchParams := map[string]string{
+		"anns_field": annsClause.Column.Name,
```
Comment (czs007) [inline internal/mysqld/executor/executor.go:153]: make these three string literals  to be CONST.
currently . package proxy defines 
	AnnsFieldKey     = "anns_field"
	TopKKey          = "topk"
        OffsetKey        = "offset"
may be we can move these to common package ?

**Pull request merged**